zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -442,15 +644,34 @@ private int getRiskId(JSONObject parameters) throws ApiException {
         try {
             riskId = Integer.parseInt(riskIdParam);
         } catch (NumberFormatException e) {
-            throwInvalidRiskId();
+            throwInvalidAlertElementId(
+                    Constant.messages.getString(
+                            "alert.api.view.alerts.param.riskId.illegal", PARAM_RISK));
         }
 
         if (riskId < Alert.RISK_INFO || riskId > Alert.RISK_HIGH) {
-            throwInvalidRiskId();
+            throwInvalidAlertElementId(
+                    Constant.messages.getString(
+                            "alert.api.view.alerts.param.riskId.illegal", PARAM_RISK));
         }
         return riskId;
     }
 
+    private RecordHistory getRecordHistory(TableHistory tableHistory, Integer id)
```

Comment:
I ripped this code from CoreAPI (instead of making that method `public static`, I'm happy to do it another way if applicable. (Or we could move it to some sort of APIUtils class?)